### DISCUSSION TITLE
Getting a Flatline RoR in 2.10.0 at 30F/min every roast

### DISCUSSION THREAD

**Discussion by oldsparks** [artisan-roaster-scope/artisan#1581] [General]
So I just started seasoning roasts on a Yoshan DY-2kg roaster. Roaster uses MODBUS over USB. This is the first time I've used Artisan over Modbus but I haven't seen this before with over 450 roasts using the same software with a Phidget setup. Running on Win11 laptop.

On each of my four seasoning roasts I got a flatline RoR when it hit 30F/min. The duration was not always the same but the trigger seemed to be at 30F every time. Duration varied from about 30s to almost 50s and then always continued without issues to the end of the roast. 

The BT continues to move in a normal fashion so I don't suspect comms. Has anyone experienced this? Is it a glitch in Artisan? I searched for "flatline" but didn't get any hits.

**Comment by MAKOMO** [comment]
As your BT signal is fine, this is not a communication issue. The RoR is computed from the BT. Maybe you have some limits set on your RoR signal. Check using menu Config >> Curves, 2nd tab "Filters" in the Rate of Rise Filter section. Either increase the max limit or deactivate that filter completely.

**Reply by oldsparks** [reply]
I've removed everything post roast. Plenty of noise but the flat line is still there. No filters. Anomaly at 3:35 to 4:35. And even with filters or ranges on, 30F/min is well within the min-max of the ranges of a normal roast, so it should be clipped the way it is. It's really odd. 

Please excuse the ugly graph, only 4 roasts with this roaster and I appreciate any insight/help. 

![Seasoning Roast](https://github.com/artisan-roaster-scope/artisan/assets/172350143/cf8c414f-88b1-4672-9d67-af00bcb97e39)

**Comment by MAKOMO** [comment]
Maybe you want to share more information which allows to investigate this further, like which version of Artisan you are using and maybe you can attach that *.alog profile here (rename it to end in .txt and drag-and-drop it here on a GitHub comment field like you did with that image).

The noise is in the signal (Artisan is not producing any noise) and it looks like [digital noise](https://artisan-roasterscope.blogspot.com/2017/06/digital-noise.html). Most likely your machine is returning the data without any decimal. Not a MODBUS problem (which is just a protocol to transport data), but an issue with the telemetry of your machine.

**Reply by oldsparks** [reply]
Thanks for pointing me the right direction. I really appreciate that. When I took all the filters off and then decreased the span to 1 or 2 seconds it looked like clipping to me. But clipping what? Now it makes perfect sense. Unfortunately I've figured out the Hanyoung Nux VX4's which is what Yoshan provided in this roaster do not output decimal data. Apparently other models in this line have the ability for at least one decimal place, but of course the form factor is different and they won't fit in the panel. I may have to shop for a 48x48mm controller with 2 decimal place output. 

I was so busy with the larger issues of working with my first drum roaster and trying to control the roast well into second cracks without starting a fire I completely missed that there was no decimal displays on the ET&BT LCDs. If you know of a 48x48mm controller that would fit the bill, or any other options in Artisan I can use other than increasing the span and rounding I'd appreciate the tips.
